cartography/intel/gsuite/api.py

```diff
@@ -0,0 +1,243 @@
+    groups = []
+    while request is not None:
+        try:
+            resp = request.execute()
+        except HttpError as e:
+            logger.warning('HttpError occurred in api.get_all_groups(), returning empty list. Details: %r', e)
+            groups = []
+            break
+        groups = groups + resp.get('groups', [])
+        request = admin.groups().list_next(request, resp)
+    return groups
+
+
+def get_all_groups_for_email(admin, email):
+    """ Fetch all groups of which the given group is a member
+
+    Arguments:
+        email: A string representing the email address for the group
+
+    Returns a list of Group models
+    Throws GoogleException
+    """
+    request = admin.groups().list(userKey=email, maxResults=500)
+    groups = []
+    while request is not None:
+        try:
+            resp = request.execute()
+        except HttpError as e:
+            logger.warning('HttpError occurred in api.get_groups_for_email(), returning empty list. Details: %r', e)
+            groups = []
+            break
+        groups = groups + resp.get('groups', [])
+        request = admin.groups().list_next(request, resp)
+    return groups
+
+
+def get_members_for_group(admin, group_email):
+    """ Get all members for a google group
+
+    :param group_email: A string representing the email address for the group
+
+    :return: List of dictionaries representing Users or Groups.
+    """
+    request = admin.members().list(
+        groupKey=group_email,
+        maxResults=500,
+    )
+    members = []
+    while request is not None:
+        try:
+            resp = request.execute()
+        except HttpError as e:
+            logger.warning('HttpError occurred in api.get_members_for_group(), returning empty list. Details: %r', e)
+            members = []
+            break
+        members = members + resp.get('members', [])
+        request = admin.members().list_next(request, resp)
+
+    return members
+
+
+def get_all_users(admin):
+    """
+    Return list of Google Users in your organization
+    Returns empty list if we are unable to enumerate the groups for any reasons
+    https://developers.google.com/admin-sdk/directory/v1/guides/manage-users
+
+    :param admin: apiclient discovery resource object
+    see
+    :return: List of Google users in domain
+    see https://developers.google.com/admin-sdk/directory/v1/guides/manage-users#get_all_domain_users
+    """
+    request = admin.users().list(customer='my_customer', maxResults=500, orderBy='email')
+    users = []
+    while request is not None:
+        try:
+            resp = request.execute()
+        except HttpError as e:
+            logger.warning('HttpError occurred in api.get_all_users(), returning empty list. Details: %r', e)
+            users = []
+            break
+        users = users + resp.get('users', [])
+        # break
+        request = admin.users().list_next(request, resp)
+    return users
+
+
+def get_ingestion_groups_qry():
```

Comment:
No need to factor this out to its own function.  I would keep this in the `load` for readability.

Reply:
Yeah I went back and forth on this -- I put this in its own function so it'd be easier to unittest:
https://github.com/lyft/cartography/pull/126/files/d636154174794d9851924c4a8664f62403539cbc#diff-c8a41175b54e8c80d326ce5f8a5e4da4R123

Otherwise, if you modified the query, then you'd need to modify two files (the unit test as well).

I'm ambivalent towards both and I can revert it back into it's own function.